client/state/products-list/selectors/compute-full-and-monthly-prices-for-plan.ts

```diff
@@ -66,19 +95,16 @@ function computePricesForWpComPlan( state, siteId, planObject ) {
 }
 
 /**
- * Compute the Jetpack Search product tiered price based on the site's number of posts.
- *
- * @param {object} state Current redux state
- * @param {number} siteId Site ID to consider
- * @param {object} productSlug The product slug
- * @returns {number} The tiered price (given the site's number of published posts)
+ * Compute the Jetpack Search product tiered price based on the site's number of published posts.
  */
-function getSearchProductTierPrice( state, siteId, productSlug ) {
+function getSearchProductTierPrice( state: AppState, siteId: number, productSlug: string ): number {
 	const postsCounts = getAllPostCounts( state, siteId, 'post' );
 	const postsCount = postsCounts?.publish || 0;
 	const priceTierList = getProductPriceTierList( state, productSlug );
 	const tier = priceTierList.filter(
-		( tierItem ) => postsCount >= tierItem.minimum_units && postsCount <= tierItem.maximum_units
```

Comment:
Since not all tiers have a maximum, I think this was working previously only because of some implicit conversion of `undefined` to 0.